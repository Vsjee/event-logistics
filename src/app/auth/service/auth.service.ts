import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root',
})
export class AuthService {
  capitalizeText(text: string): string {
    return text[0].toUpperCase() + text.substring(1);
  }
}
