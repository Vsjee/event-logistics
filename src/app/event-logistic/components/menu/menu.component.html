<header class="header" (mouseleave)="disabledPopup(false)">
  <picture class="header__logo">
    <img
      src="../../../../assets/chargebee.svg"
      alt="Logo"
      class="header__img"
    />
  </picture>
  <nav class="header__nav">
    <ul class="header__list">
      <li
        class="header__item"
        *ngFor="let item of menu"
        [routerLink]="item.route"
        routerLinkActive="activeRoute"
      >
        {{ item.name }}
      </li>
    </ul>
  </nav>

  <div class="header__contain">
    <button
      title="Favoritos"
      class="header__favorites"
      (click)="openFavorites()"
    >
      <img
        src="../../../../assets/icons/star.svg"
        alt=""
        class="header__icon"
        [ngStyle]="{ 'width.px': '15' }"
      />
      <span>Favoritos</span>
    </button>
    <button class="header__user" (mouseenter)="openPopup()">
      <img
        src="https://images.unsplash.com/photo-1539571696357-5a69c17a67c6?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=387&q=80"
        alt=""
        class="header__img header__img--width"
      />
    </button>
    <app-popup-user *ngIf="visiblePopup" (listeningPopup)="disabledPopup($event)"></app-popup-user>
  </div>
</header>
