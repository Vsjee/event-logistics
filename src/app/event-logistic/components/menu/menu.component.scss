.header {
  width: 100%;
  height: 80px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 15px;
  position: fixed;
  top: 0;
  background-color: var(--bg);
  z-index: 1;
}

.header__logo {
  display: inline-block;
  width: 100px;
}

.header__img {
  width: 100%;
  height: 100%;
}

.header__user {
  display: inline-block;
  width: 39px;
  height: 39px;
  border: none;
  background-color: var(--bg);
  cursor: pointer;

  .header__img--width {
    object-fit: cover;
    border-radius: 50%;
  }
}

.header__list {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 10px;
  list-style: none;

  .header__item {
    font-size: var(--font-size-sm);
    font-weight: var(--font-weight-light);
    cursor: pointer;
    border-bottom: 3px solid transparent;
    transition: 0.3s ease;
  }

  .header__item:hover {
    border-color: var(--primary-color);
  }
}

.activeRoute {
  border-color: var(--primary-color) !important;
}

.header__contain {
  position: relative;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.header__favorites {
  border: none;
  background-color: var(--bg);
  cursor: pointer;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-size: var(--font-size-sm);
  padding-right: 8px;
}
