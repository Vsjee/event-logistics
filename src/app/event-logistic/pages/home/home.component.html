<main class="home">
  <section class="welcome">
    <article class="welcome__article">
      <h1 class="welcome__text">Los eventos para hoy</h1>
      <p class="welcome__date">{{ date | date : "fullDate" }}</p>
    </article>
    <article class="welcome__event">
      <app-card-event [cardItem]="cardItem"></app-card-event>
    </article>
  </section>

  <section class="next">
    <h1 class="next__title">Pr√≥ximos Eventos</h1>
    <article class="next__event">
      <app-card-event
        *ngFor="let item of nextEvent"
        [cardItem]="item"
      ></app-card-event>
    </article>
  </section>
</main>


<app-favorites></app-favorites>
