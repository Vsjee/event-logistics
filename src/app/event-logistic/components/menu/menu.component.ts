import { Component } from '@angular/core';

import { MenuItem } from 'src/app/interfaces/event-logistic.interface';
import { HomeService } from '../../services/home.service';

@Component({
  selector: 'app-menu',
  templateUrl: './menu.component.html',
  styleUrls: ['./menu.component.scss'],
})
export class MenuComponent {
  menu: MenuItem[] = [
    {
      name: 'Inicio',
      route: '/eventos/manage/inicio',
    },
    {
      name: 'Crear Evento',
      route: '/eventos/manage/crear',
    },
    {
      name: 'Ver eventos',
      route: '/eventos/manage/ver',
    },
  ];

  visiblePopup: boolean = false;
  visibleFavorites: boolean = false;

  constructor(private homeService: HomeService) {}

  ngOnInit(): void {}

  openPopup(): void {
    this.visiblePopup = true;
  }

  disabledPopup(): void {
    this.visiblePopup = false;
  }

  openFavorites(): void {
    this.homeService.openFavorites$.next(true);
  }
}
