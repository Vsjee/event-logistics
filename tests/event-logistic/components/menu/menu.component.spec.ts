import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { MenuComponent } from '../../../../src/app/event-logistic/components/menu/menu.component';
import { HomeService } from '../../../../src/app/event-logistic/services/home.service';

describe('MenuComponent', () => {
  let component: MenuComponent;
  let fixture: ComponentFixture<MenuComponent>;
  let compiled: HTMLElement;
  let service: HomeService;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [MenuComponent],
      imports: [RouterTestingModule],
      providers: [HomeService],
    }).compileComponents();

    fixture = TestBed.createComponent(MenuComponent);
    service = TestBed.inject(HomeService);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
    jest.clearAllMocks();
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe crear el snapshot inicial', () => {
    expect(compiled).toMatchSnapshot();
  });

  test('debe de abrir el popup de usuario', () => {
    component.openPopup();
    expect(component.visiblePopup).toBeTruthy();
  });

  test('debe de cerrar el popup de usuario', () => {
    component.disabledPopup();
    expect(component.visiblePopup).toBeFalsy();
  });

  test('debe enviar true para abrir la ventana de favoritos', () => {
    component.openFavorites();
    fixture.detectChanges();
    jest.spyOn(service.openFavorites$, 'next');

    expect(service.openFavorites$.next).toBeTruthy();
  });
});
