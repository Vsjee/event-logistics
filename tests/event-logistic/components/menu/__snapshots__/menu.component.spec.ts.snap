// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuComponent debe crear el snapshot inicial 1`] = `
<div
  id="root1"
>
  <header
    class="header"
  >
    <picture
      class="header__logo"
    >
      <img
        alt="Logo"
        class="header__img"
        src="../../../../assets/chargebee.svg"
      />
    </picture>
    <nav
      class="header__nav"
    >
      <ul
        class="header__list"
      >
        <li
          class="header__item"
          routerlinkactive="activeRoute"
          tabindex="0"
        >
           Inicio 
        </li>
        <li
          class="header__item"
          routerlinkactive="activeRoute"
          tabindex="0"
        >
           Crear Evento 
        </li>
        <li
          class="header__item"
          routerlinkactive="activeRoute"
          tabindex="0"
        >
           Ver eventos 
        </li>
        
      </ul>
    </nav>
    <div
      class="header__contain"
    >
      <button
        class="header__favorites"
        title="Favoritos"
      >
        <img
          alt=""
          class="header__icon"
          src="../../../../assets/icons/star.svg"
          style="width: 15px;"
        />
        <span>
          Favoritos
        </span>
      </button>
      <button
        class="header__user"
      >
        <img
          alt=""
          class="header__img header__img--width"
          src="https://images.unsplash.com/photo-1539571696357-5a69c17a67c6?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=387&q=80"
        />
      </button>
      
    </div>
  </header>
</div>
`;
